src/Capabilities/ICanHandleRequests.php

```diff
@@ -30,11 +31,12 @@
 interface ICanHandleRequests
 {
 	/**
-	 * @param array $request The $_REQUEST content (including content from the PHP input stream)
+	 * @param ModuleHTTPException $httpException The special HTTPException Module in case of underlying errors
+	 * @param array               $request       The $_REQUEST content (including content from the PHP input stream)
 	 *
 	 * @return ResponseInterface responding to the request handling
 	 *
 	 * @throws HTTPException\InternalServerErrorException
 	 */
-	public function run(array $request = []): ResponseInterface;
+	public function run(ModuleHTTPException $httpException, array $request = []): ResponseInterface;
```

Comment:
@MrPetovan jfyi

I'm really not happy about extending the `run()`. I think it's a design flaw and leads to a general design flaw I introduced with `BaseModule`.

I think rather than executing `run()` from inside `BaseModule`, including all the `run()` depending methods, there should be a `ModuleExecutor` class (just a working title)  and so all these dependencies in the constructor will almost vanish, so new Modules won't have to add these 7 dependencies in the constructor and the `run()` won't need to be overridden with new Modules..

--> followup

I already started, but it would extend this PR way too much

Reply:
Would this refactoring take place before or after this PR?

Reply:
After :-)
Step by step